base/datatype_reference.bal

```diff
@@ -78,3 +78,71 @@ public type Reference record {|
     Identifier identifier?;
     string display?;
 |};
+
+# Create a relative Reference to FHIR Resource.
+#
+# + referenceResourceModel - FHIR Resource model of the Reference
+# + relativePathToReference - Relative path to Reference
+# + return - FHIRError if the Reference resource type is not defined, else FHIR Reference
+public function createRelativeFhirReference(typedesc<anydata> referenceResourceModel, string relativePathToReference) returns Reference|FHIRError {
+    ResourceDefinitionRecord? def = referenceResourceModel.@ResourceDefinition;
+    string? resourceType = def?.resourceType;
+
+    if resourceType is string {
+        return {
+            reference: string `${resourceType}/${relativePathToReference}`,
+            'type: resourceType
+        };
+    } else {
+        return createFHIRError("Resource type of the Reference resource model is not defined", ERROR, PROCESSING, errorType = PROCESSING_ERROR);
+    }
```

Comment:
```suggestion
    }
    return createFHIRError("Resource type of the Reference resource model is not defined", ERROR, PROCESSING, errorType = PROCESSING_ERROR);
```

Reply:
check other places as well

Reply:
Fixed here and other applicable places in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/76360f623a5442dff836e7d851df34125a5818f8